clients/src/main/java/org/apache/kafka/common/record/AbstractRecordBatch.java

```diff
@@ -33,4 +36,25 @@ public boolean isCompressed() {
         return compressionType() != CompressionType.NONE;
     }
 
+    static CloseableIterator<Record> wrapAsCloseable(Iterator<Record> inner) {
```

Comment:
This seems generally useful for `CloseableIterator`, not just in the context of records. Perhaps we can move it there and use `CloseableIterator.wrap(anotherIterator)`?

Reply:
Ack.